documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+|No default
+|(Optional) Username of the database which will be used to access the database storage.
+
+|[[deprecated-jdbc-offset-password]]<<deprecated-jdbc-offset-password, `offset.storage.jdbc.password`>>
+|No default
+|(Optional) Password of the database which will be used to access the database storage.
+
+|[[deprecated-wait-retry-delay-ms]]<<deprecated-wait-retry-delay-ms, `offset.storage.jdbc.wait.retry.delay.ms`>>
+|3 seconds
+|(Optional) Delay of retry on wait for connection failure.
+
+|[[deprecated-retry-max-attempts]]<<deprecated-retry-max-attempts, `offset.storage.jdbc.retry.max.attempts`>>
+|5
+|(Optional) Maximum number of retry attempts before giving up.
+
+|[[deprecated-jdbc-offset-table-name]]<<deprecated-jdbc-offset-table-name, `offset.storage.jdbc.offset.table.name`>>
+|`debezium_offset_storage`
+|The name of the table to which the offsets will be stored.
+
+|[[deprecated-jdbc-offset-table-ddl]]<<deprecated-jdbc-offset-table-ddl, `offset.storage.jdbc.offset.table.ddl`>>
+|<<jdbc-offset-table-ddl-default, `see table section`>>
+|DDL statement to create the offset table.
+
+|[[deprecated-jdbc-offset-table-select]]<<deprecated-jdbc-offset-table-select, `offset.storage.jdbc.offset.table.select`>>
+|<<jdbc-offset-table-ddl-default, `see table section`>>
+|DML statement to read offsets stored from the table.
+
+|[[deprecated-jdbc-offset-table-insert]]<<deprecated-jdbc-offset-table-insert, `offset.storage.jdbc.offset.table.insert`>>
+|<<jdbc-offset-table-select-default, `see table section`>>
+|DML statement to write offsets to the table.
+
+|[[deprecated-jdbc-offset-table-delete]]<<deprecated-jdbc-offset-table-delete, `offset.storage.jdbc.offset.table.delete`>>
+|<<jdbc-offset-table-insert-default, `see table section`>>
+|DML statement to remove offsets from the table.
+
+|===
+
+=== Offset Table defaults
+
+[[jdbc-offset-table-ddl-default]]<<jdbc-offset-table-ddl-default, 1. Table Structure>>
+[SQL]
+----
+CREATE TABLE %s (
+id VARCHAR(36)      NOT NULL,
+offset_key          VARCHAR(1255),
+offset_val          VARCHAR(1255),
+record_insert_ts    TIMESTAMP NOT NULL,
+record_insert_seq   INTEGER NOT NULL)
+----
+
+[[jdbc-offset-table-select-default]]<<jdbc-offset-table-select-default, 2. Select query>>
+[SQL]
+----
+SELECT id, offset_key, offset_val FROM %s ORDER BY record_insert_ts, record_insert_seq
+----
+
+[[jdbc-offset-table-insert-default]]<<jdbc-offset-table-insert-default, 3. Insert query>>
+[SQL]
+----
+INSERT INTO %s(id, offset_key, offset_val, record_insert_ts, record_insert_seq)
+    VALUES ( ?, ?, ?, ?, ? )
+----
+
+[[jdbc-offset-table-delete-default]]<<jdbc-offset-table-delete-default, 4. Delete query>>
+[SQL]
+----
+DELETE FROM %s
+----
+
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[jdbc-history-type]]<<jdbc-history-type, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.jdbc.history.JdbcSchemaHistory`.
+
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?